### DISCUSSION TITLE
How about side-bar-in-window ?

### DISCUSSION THREAD

**Discussion by rageworx** [fire-eggs/YAIV#81] [Ideas]
Dear Kevin,

I like to show some old application, has left side toolbar like (menu ?) ... I believe you may be interested.
This application was made in about 5 years ago, working with CUDA8 on windows and FLTK/MinGW-W64.

![image](https://user-images.githubusercontent.com/6948225/125872831-a2382db0-1095-4f5b-9719-a659a91421d8.png)

There's a UDP server ( yes, it is FLTK ) that loads Tomography slices ( DICOM/DCM tag files )
And a reconstructing & rendering server with FLTK client application.

These applications working each together in same or different network or local.

![image](https://user-images.githubusercontent.com/6948225/125872971-f6db6d23-2a6b-4510-8c33-b3616bf94bdb.png)

![image](https://user-images.githubusercontent.com/6948225/125872987-625562e7-496a-4a8b-a889-f20bcad47b8c.png)

As you can see, there's a vertical black left-sided tool bar on client application.
It dynamically resized as like this,

![image](https://user-images.githubusercontent.com/6948225/125873046-a1e65e06-5e54-4fd5-b59d-29dacdf748ec.png)

And each images are an implemented Fl_Button ( maybe it was my Fl_Image_Button, as I remembered ... )

![image](https://user-images.githubusercontent.com/6948225/125873141-c774e1e0-c5fb-471b-a269-763ffab27bc0.png)

![image](https://user-images.githubusercontent.com/6948225/125873178-99c62e63-cee9-48b8-a544-2187c4251857.png)

![image](https://user-images.githubusercontent.com/6948225/125873199-b11e834b-e90f-4087-8baf-a8298cdb514f.png)

This application was made with some patented commercial license ( specially in CUDA processing ), so I can't share all source codes but client application source may believe to be shared.

How about this type of side bar ?
I think detaching toolbar as like legacy Photoshop may not convenient style today ( modern Photoshops are toolbars in application window region and not detachable )

**Comment by fire-eggs** [comment]
Very nice!

Still pondering options. Some of the issues I've yet to fix could be solved by not making the toolbar floating. 

To show you the current state of things, here's a screen cap:

https://drive.google.com/file/d/1PBkM1fPeMhLtbz0qOsqOFk07-jb_O1Qz/view?usp=sharing

A horizontal (docked to the top) toolbar is also possible.

The video also shows the glitching that takes place during resize of the 'danbooru' window.

**Reply by rageworx** [reply]
That your tool-bar is much better than I expected.
And some flashing - that maybe belong to occured while resizing method.
I like to know what is that toolbar to fix what affects flashing visual.
Regards, Raph.K.

**Reply by fire-eggs** [reply]
At this point I'm convinced a drag-able, floating toolbar is a fun thing to play with but not practical. I'll go for a fixed toolbar, but the user needs to have the choice of docked top/bottom/left/right [or no toolbar].

Floating windows still make sense to me for features like Danbooru, metadata, and "extra" things like animation playback controls. But they won't be dock-able.
